aiosqlite/cursor.py

```diff
@@ -12,18 +12,30 @@ class Cursor:
     def __init__(self, conn: "Connection", cursor: sqlite3.Cursor) -> None:
         self._conn = conn
         self._cursor = cursor
+        self.__chunk: Optional[Iterator[sqlite3.Row]]
+        self.iter_chunk_size = 1024
 
     def __aiter__(self) -> "Cursor":
         """The cursor proxy is also an async iterator."""
+        self.__chunk = None
         return self
 
     async def __anext__(self) -> sqlite3.Row:
         """Use `cursor.fetchone()` to provide an async iterable."""
-        row = await self.fetchone()
-        if row is None:
+        if self.__chunk is None:
+            self.__chunk = await self.__fetch_anext_chunk()
+
+        try:
+            return self.__chunk.__next__()
+        except StopIteration:
+            self.__chunk = await self.__fetch_anext_chunk()
+            return self.__chunk.__next__()
+
+    async def __fetch_anext_chunk(self) -> Iterator[sqlite3.Row]:
+        chunk = await self.fetchmany(self.iter_chunk_size)
+        if not chunk:
             raise StopAsyncIteration
-
-        return row
+        return iter(chunk)
```

Comment:
Overall, I think all of this could be more cleanly implemented using an async generator. This removes the need for `self.__chunk`, and also reduces the complexity around handling/raising stop iteration, etc. Something like:

```python
def __aiter__(self):
    return self._fetch_chunked()

async def _fetch_chunked(self):
    while True:
        rows = await self.fetchmany(...)
        for row in rows:
            yield row
```

Reply:
Indeed way better this way, also performance wise. Sometimes it's hard to remember all this pythonic coolness coming from a different language.